New sidebar and website content overhaul

HereAround opened the pull request.

HereAround committed:
New sidebar

HereAround committed:
Update content of website to new sidebar

aaruni96 committed:
Make collapsible header linkable to content

aaruni96 commented:
Made it so that 'getting started' can link to some content, and the arrow next to it does the expand/collapse for the sub items.

aaruni96 committed:
add back the small footer on the sidebar

HereAround committed:
More improvements

HereAround commented:
@lgoettgens , I have just pushed a few more changes here. Roughly speaking, the idea is to reorganize the structure of the website into logical building blocks (whereas there is a bit a of a jungle on the current website). But of course, eventually also adjustments to at least some of the content would have to be added. And some pages are not yet inserted, leading to a couple 404-errors.

As such: Currently a first rough draft. Let me know what you think about this!

cc @aaruni96 @fingolfin

lgoettgens commented:
Is there a way to get a preview of the webpage or would I need to build locally?

HereAround commented:
> Is there a way to get a preview of the webpage or would I need to build locally?

There is a way for this. I pushed the changes to the gh-pages branch of my fork (https://github.com/HereAround/oscar-website/tree/gh-pages), and somehow it was possible to get a preview from this earlier today. I believe there was a link "hidden" somewhere, but I cannot find it anymore. @aaruni96 could you remind me?

HereAround commented:
Finally found it @lgoettgens : https://herearound.github.io/oscar-website/

lgoettgens commented:
> Finally found it : [herearound.github.io/oscar-website](https://herearound.github.io/oscar-website/)

I like the new sidebar structure. Seems a lot less bloated and more usable now.
maybe @gfourier wants to have a look as well to see if this is going in the direction he imagined?

fingolfin commented:
Unfortunately expanding the submenus still doesn't work for me in Safari (as discussed on Wednesday). 

Also some users have JavaScript turned off. So IMHO at the very least, if JavaScript is off, the menus should be fully expanded so that one can reach everything.

Even better would be if the menus were fully CSS driven. This could probably achieved using `<details>` and `<summary>` plus suitable CSS styling. See e.g. [this page](https://www.sitepoint.com/style-html-details-element/) for lots of nice styling suggestions.

Or maybe just always should them all expanded...

HereAround committed:
Add missing pages

Fill in details to index page on getting-started

HereAround committed:
Move contact details from installation to getting-in-touch

HereAround committed:
Move "Getting a gap prompt" away from the installation page

HereAround committed:
Move link to external tutorials from install page to tutorials page

HereAround committed:
Make it clearer what IJulia is used for

HereAround committed:
Overhaul getting-started

HereAround committed:
Move up "Getting-in-touch"

HereAround committed:
More improvements

HereAround committed:
Add index.md content for credits and citations

HereAround committed:
Fix broken link

HereAround committed:
Fix broken link

HereAround committed:
Overhaul main landing page

HereAround committed:
Include mardi into software credits

HereAround committed:
Make collapsable menus work with css

HereAround committed:
Minimize css changes

HereAround committed:
Prevent submenu from collapsing when navigating to submenu link

HereAround committed:
Make submenus not bold by default

HereAround committed:
Remove unused file

HereAround committed:
Fix accidental submenu collapse

HereAround committed:
Place list of papers citing OSCAR into yml file

HereAround committed:
Overhaul installation pages

HereAround committed:
Overhaul tutorial page

HereAround committed:
Overhaul documentation

HereAround committed:
Improve contributors page

HereAround committed:
Improve installation instructions

HereAround committed:
Improve tutorials page

HereAround commented:
> Unfortunately expanding the submenus still doesn't work for me in Safari (as discussed on Wednesday).
> 
> Also some users have JavaScript turned off. So IMHO at the very least, if JavaScript is off, the menus should be fully expanded so that one can reach everything.
> 
> Even better would be if the menus were fully CSS driven. This could probably achieved using `<details>` and `<summary>` plus suitable CSS styling. See e.g. [this page](https://www.sitepoint.com/style-html-details-element/) for lots of nice styling suggestions.
> 
> Or maybe just always should them all expanded...

Thank you for the pointer. I played with it more, and I believe the submenus should now work for you @fingolfin . Please try out and let me know.

As you may notice, I have also re-grouped the content of certain pages. For instance, we anticipate to display installation instructions only for the OS used to open the webpage. While I could not yet set this up, I have grouped instructions for each OS on a separate page. Hopefully, this makes the final transition easy.

Also, overhaul of the main landing page (including the "Get OSCAR now" section, that you were looking for).

There is a minor problem with the tutorials page (clicking any tutorial group selector icon twice gets us a wrong path, likely due to some mistake in the logic of the page which I could not yet find - but maybe @aaruni96 find the error quickly). Modulo this shortcoming, I believe this is ready for review.

aaruni96 committed:
Minor tweak to sidebar/collapsible

fingolfin commented:
Looking better and better! Some suggestions:
- if I click on e.g. "Events" (which sends me to https://herearound.github.io/oscar-website/events/) I really think the items under "Events" should be made visible
  - however overall I am a bit concerned about the discoverability of these submenus... perhaps we can experiment with some alternatives and/or talk about it next week
  - using triangles on the left (as "bullet points") *and* on the right (to indicate a submenu can be extended is not optimal; perhaps regular (round) bullet points can be used)
  - or perhaps icons (like on the front page), a fantastic source for those is <https://icon-sets.iconify.design>
  - perhaps we could also investigate horizontal navigation at the top, e.g. <https://www.w3schools.com/howto/howto_css_subnav.asp> -- but I think that goes to far for now, such a change should probably part of a full website redesign
- the triangles-on-the-right could perhaps be right aligned?

Anyway it's probably easier to discuss this in person next week

aaruni96 commented:
> if I click on e.g. "Events" (which sends me to https://herearound.github.io/oscar-website/events/) I really think the items under "Events" should be made visible

working on this right now

> the triangles-on-the-right could perhaps be right aligned?

also this

quick preview while I get some more things ready :

![image](https://github.com/user-attachments/assets/231ee198-ef76-485b-9448-c7d1ea48ccec)

aaruni96 committed:
fix tutorials

aaruni96 committed:
autodetect install platform

aaruni96 commented:
@HereAround can you merge this to your gh-pages again so that its available for preview ?

HereAround commented:
Thank you @aaruni96 . Just updated my gh-pages branch.

aaruni96 committed:
more changes related to install / autodetect

aaruni96 committed:
Restore unintended deletions from tutorial page

Also, add slightly more clever redirecting based on OS

HereAround committed:
Improve wording on tutorial page

HereAround committed:
Overhaul main page of getting-started

HereAround committed:
Remove "WSL" from Mac and Linux install instructions

HereAround committed:
Remove unused file

HereAround committed:
Improvements to main landing page

HereAround committed:
Add link to contact and support to installation page

HereAround commented:
Just pushed more changes. I believe we got at least most of the features covered that we wanted. In particular, thanks to the changes by @aaruni96 , the installation page will now display instructions specific to the OS used to open the website.

@fingolfin When you find the time, take a look at https://herearound.github.io/oscar-website/ and let us know what more changes are desired before we merge this PR.

HereAround commented:
Pinging also @lgoettgens and @gfourier.

fingolfin commented:
Thank you this is shaping up very nicely! Just had a look together with @HereAround and wanted to record what I said:
- the website is responsive and changes its layout when viewed in a very narrow window (i.e.: on a smartphone). This is somewhat broken with the hierarchical navigation :-/. A possible solution would be to switch to a "burger menu" style navigation, i.e. still use a navbar (as when using the wide window) but hide it by default in narrow view and show it when the user clicks a burger menu. But this requires bigger changes to the theme...
- I like how GitHub handles pages with OS variants: they show "tabs" one can use to switch, and I think it doesn't even reload (see <https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository?platform=mac&tool=webui> for an example). Perhaps we could do something similar -- but that's not meant to hold up merging


(... there was more when I started writing this on Friday but then I got interrupted... submitting this fragment now so that it doesn't get lost. I'll have another look at the page and add more comments if I can remember them)

fingolfin commented:
Once we are sufficiently happy we should probably also share the link to the preview on Slack and ask for feedback there?

aaruni96 committed:
Update supported OSs

fingolfin commented:
It works now in my version of Safari, good!

Some more notes:
- https://herearound.github.io/oscar-website/getting-started/miscellaneous/ can probably just go for now -- or perhaps be integrated into a tutorial, or the OSCAR manual section https://docs.oscar-system.org/dev/General/other/#Notes-for-GAP-users
- perhaps rename https://herearound.github.io/oscar-website/credits/OSCAR-credits/ to "Bibliography?
- then perhaps rename "People Credits" and "Software Credits" to just "People" and "Software" ?

HereAround committed:
Remove miscellaneous

HereAround committed:
Rename OSCAR credits into bibliography

HereAround committed:
People credit/Software credit -> People/Software

HereAround committed:
Explain contributions to source code first, then tutorial contribution

HereAround committed:
Add burger menu

HereAround committed:
Automatically adjust width of "getting-started" picture to screen in question

HereAround committed:
Improvements to events section

HereAround committed:
Add tabbed navigation UI for OS selection in installation instructions

HereAround commented:
> It works now in my version of Safari, good!
> 
> Some more notes:
> 
>     * https://herearound.github.io/oscar-website/getting-started/miscellaneous/ can probably just go for now -- or perhaps be integrated into a tutorial, or the OSCAR manual section https://docs.oscar-system.org/dev/General/other/#Notes-for-GAP-users
> 
>     * perhaps rename https://herearound.github.io/oscar-website/credits/OSCAR-credits/ to "Bibliography?
> 
>     * then perhaps rename "People Credits" and "Software Credits" to just "People" and "Software" ?

I have taken care of this, added a burger menu for mobile devices (including some fine tuning of spaces), added a tabbed-nagivation UI to the installation page for selecting the OS and made a few more minor changes to the content.

I am happy with the status this is in right now (see https://herearound.github.io/oscar-website/). Will wait a few more days to gather feedback before merging.

HereAround committed:
On getting-started page, position "Tutorials" to the left

HereAround committed:
Remove page with selected talks on OSCAR

fingolfin reviewed (approved):
This is a great improvement, thank you!
